tests/test_bukuDb.py

```diff
@@ -848,6 +848,30 @@ def test_add_rec_fetch(bukuDb, caplog, fetch, url_redirect, tag_redirect, tag_er
     assert _tagset(rec.tags) == _tags
 
 
+@pytest.mark.parametrize('status, tags_fetched, tags_in, tags_except, expected', [
+    (None, None, 'foo,qux,foo bar,bar,baz', 'except,bar,foo,', ',baz,foo bar,qux,'),
+    (200, '', 'foo,qux,foo bar,bar,baz', None, ',bar,baz,foo,foo bar,qux,'),
+    (200, 'there,have been,some,tags,fetched', None,
+     'except,bar,tags,there,foo', ',fetched,have been,some,'),
+    (200, 'there,have been,some,tags,fetched', 'foo,qux,foo bar,bar,baz',
+     'except,bar,tags,there,foo', ',baz,fetched,foo bar,have been,qux,some,'),
+    (404, None, 'foo,foo bar,qux,bar,baz', 'except,bar,foo', ',baz,foo bar,http:error,qux,'),
+    (301, 'there,have been,some,tags,fetched', 'foo,foo bar,qux,bar,baz',
+     'except,bar,tags,there,foo', ',baz,fetched,foo bar,have been,http:redirect,qux,some,'),
+    (308, 'there,have been,some,tags,fetched', 'foo,foo bar,qux,bar,baz',
+     'except,http:redirect,bar,tags,there,foo', ',baz,fetched,foo bar,have been,qux,some,'),
+])
```

Comment:
Verifying various combinations of supplied tag arguments, along with fetch result (`None` means `fetch=False`)

Tag override doesn't apply to network test tags (these are opt-in anyway), but they can be filtered out via `tags_except` (e.g. `tags_except='http:500'` would allow to exclude HTTP 500 error tags from generated output, since these tend to be transient)